### PR TITLE
Use MGTransferGlobalCoarsening inside of MGTransferGlobalRefinement

### PR DESCRIPTION
**Pull request by peterrum**
This PR shows how one could use `MGTransferGlobalRefinement` within `MGTransferGlobalCoarsening`.

To see the differences, one can put
include/exadg/solvers_and_preconditioners/multigrid/transfers/mg_transfer_global_coarsening.cpp and include/exadg/solvers_and_preconditioners/multigrid/transfers/mg_transfer_global_refinement.cpp next to each other.

```diff
47,48c46,47
<     unsigned int const level  = matrixfree->get_dof_handler().get_triangulation().n_global_levels();
<     unsigned int const degree = fe.degree;
---
>     unsigned int const level      = matrixfree->get_mg_level();
>     unsigned int const degree     = fe.degree;
69,70c68,69
<         dealii::numbers::invalid_unsigned_int,
<         dealii::numbers::invalid_unsigned_int);
---
>         fine_level.h_level(),
>         coarse_level.h_level());
80,81c79,80
<         dealii::numbers::invalid_unsigned_int,
<         dealii::numbers::invalid_unsigned_int);
---
>         fine_level.h_level(),
>         coarse_level.h_level());
```

### CODE REVIEW

File: include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp [hunk 1]
```diff
@@ -984,6 +984,8 @@ MultigridPreconditionerBase<dim, Number>::do_initialize_transfer_operators(
   dealii::MGLevelObject<std::shared_ptr<dealii::MGConstrainedDoFs>> & constrained_dofs,
   unsigned int const                                                  dof_index)
 {
+  (void)constrained_dofs;
```
Comment (nfehn) [inline include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp:987]: Realizing the changes requested here https://github.com/exadg/exadg/pull/472#issuecomment-1576734029, this function can be simplified a lot I think. I believe the present PR should already do such changes and already adjust the interface of functions if some parameters are no longer needed.

Making such adjustments will help others in discussions such as https://github.com/exadg/exadg/issues/445 and related code changes / PRs.

Comment (peterrum) [reply include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp:987]: I'll do once we decide that this is the way to go.

**Comment by peterrum** [comment]
@nfehn @kronbichler I have removed the second transfer operator.

**Review by kronbichler — APPROVED** [review]

File: include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp [hunk 2]
```diff
@@ -984,19 +983,16 @@ MultigridPreconditionerBase<dim, Number>::do_initialize_transfer_operators(
...
-  {
-    auto tmp = std::make_shared<MGTransferGlobalCoarsening<dim, MultigridNumber, VectorTypeMG>>();
-
-    tmp->reinit(matrix_free_objects, dof_index);
+  (void)constrained_dofs;
 
-    transfers = tmp;
-  }
-  else if(multigrid_variant == MultigridVariant::LocalSmoothing)
+  if(multigrid_variant == MultigridVariant::GlobalCoarsening ||
```
Comment (nfehn) [inline include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp:988]: we decided for `or` syntax recently.

**Pull request merged**